synapse/storage/data_stores/main/registration.py

```diff
@@ -1315,15 +1316,14 @@ def validate_threepid_session_txn(txn):
             )
 
             if not row:
-                raise ThreepidValidationError(400, "Unknown session_id")
+                if self._ignore_unknown_session_error:
+                    row = {"client_secret": None, "validated_at": None}
```

Comment:
We could also just raise `ThreepidValidationError, "Validation token not found or has expired"` here?

Reply:
Possibly, but I wanted to avoid duplicating code.

Reply:
Fair enough. Could you add a comment here explaining what we expect to happen when these are set to `None` then?